src/plugins/intel_gna/src/transformations/gather_sinking_utils.cpp

```diff
@@ -0,0 +1,285 @@
+        auto new_axes_const = axes_const->clone_with_new_inputs({});
+        auto new_gather = std::make_shared<Gather>(input_node, new_indices_const, new_axes_const);
+
+        main_node->input(i).replace_source_output(new_gather->output(0));
+
+        copy_runtime_info(input_node.get_node_shared_ptr(), {new_gather, new_indices_const, new_axes_const});
+
+        new_nodes.push_back(new_gather);
+    }
+
+    return new_nodes;
+}
+
+}  // namespace sink_backward
+
+namespace {
+#define CHECK_GATHER_SINKING_SUPPORTED(TYPE, node) \
+    if (dynamic_cast<TYPE*>(node)) {                  \
+        return true;                                  \
+    }
+
+bool CanPropagateGatherForwardThrough(Node* node) {
+    CHECK_GATHER_SINKING_SUPPORTED(ov::op::util::UnaryElementwiseArithmetic, node);
+    CHECK_GATHER_SINKING_SUPPORTED(Clamp, node);
+    CHECK_GATHER_SINKING_SUPPORTED(Elu, node);
+    CHECK_GATHER_SINKING_SUPPORTED(SoftPlus, node);
+    CHECK_GATHER_SINKING_SUPPORTED(LogicalNot, node);
+    CHECK_GATHER_SINKING_SUPPORTED(Convert, node);
+    return false;
+}
+
+#undef CHECK_GATHER_SINKING_SUPPORTED
+
+bool CanGatherPropagateForward(NodePtr node) {
+    for (size_t i = 0; i < node->get_output_size(); ++i) {
+        for (auto& consumer_input : node->output(i).get_target_inputs()) {
+            if (!CanPropagateGatherForwardThrough(consumer_input.get_node()))
+                return false;
+        }
+    }
+
+    return true;
+}
+
+}  // namespace
+
+void UpdateForwardGatherSinkingAbility(NodePtr node) {
+    if (!CanGatherPropagateForward(node))
+        mark_as_no_gather_sinking_node(node);
+}
+
+namespace {
+
+struct GatherInfo {
+    bool isEmpty() const { return indices.empty(); }
+    bool operator == (const GatherInfo& another) {
+        if (indices.size() != another.indices.size())
+                return false;
+        if (!std::equal(indices.begin(),
+                        indices.end(),
+                        another.indices.begin()))
+            return false;
+        return axis == another.axis;
+    }
+    bool operator != (const GatherInfo& another) {
+        return !(*this == another);
+    }
+
+    ov::AxisVector indices;
+    int64_t axis = {};
+};
+
+GatherInfo GetGatherInfo(Node* node) {
+    GatherInfo gather_info;
+
+    auto gather_node = dynamic_cast<Gather*>(node);
+    if (!gather_node)
+        return {};
+
+    auto constant_node = as_type_ptr<Constant>(gather_node->input_value(1).get_node_shared_ptr());
+    if (!constant_node)
+        return {};
+
+    gather_info.indices = constant_node->get_axis_vector_val();
+
+    constant_node = as_type_ptr<Constant>(gather_node->input_value(2).get_node_shared_ptr());
+    if (!constant_node)
+        return {};
+
+    gather_info.axis = constant_node->get_axis_vector_val()[0];
+
+    return gather_info;
+}
+
+Node* FindFirstConsumer(NodePtr node) {
+    for (size_t output_idx = 0; output_idx < node->get_output_size(); ++output_idx) {
```

Comment:
Can be changed to range for? output : outputs ?

Reply:
I see the ways, that cause more source code here. I would like not to change it

Reply:
added possible implementation, looks mode compact, but need to test